bolt/lib/Target/AArch64/CMakeLists.txt

```diff
@@ -4,6 +4,39 @@ set(LLVM_LINK_COMPONENTS
   AArch64Desc
   )
 
+if(BOLT_BUILT_STANDALONE)
+  # tablegen, copied from llvm/lib/Target/AAarch64/CMakeLists.txt
+  set(LLVM_TARGET_DEFINITIONS ${LLVM_MAIN_SRC_DIR}/lib/Target/AArch64/AArch64.td)
+  list(APPEND LLVM_TABLEGEN_FLAGS -I ${LLVM_MAIN_SRC_DIR}/lib/Target/AArch64)
+  tablegen(LLVM AArch64GenAsmMatcher.inc -gen-asm-matcher)
+  tablegen(LLVM AArch64GenAsmWriter.inc -gen-asm-writer)
+  tablegen(LLVM AArch64GenAsmWriter1.inc -gen-asm-writer -asmwriternum=1)
+  tablegen(LLVM AArch64GenCallingConv.inc -gen-callingconv)
+  tablegen(LLVM AArch64GenDAGISel.inc -gen-dag-isel)
+  tablegen(LLVM AArch64GenDisassemblerTables.inc -gen-disassembler)
+  tablegen(LLVM AArch64GenFastISel.inc -gen-fast-isel)
+  tablegen(LLVM AArch64GenGlobalISel.inc -gen-global-isel)
+  tablegen(LLVM AArch64GenO0PreLegalizeGICombiner.inc -gen-global-isel-combiner
```

Comment:
I'm not sure BOLT needs these .inc files, but I'm fine if they're included transitively

Reply:
You can remove lines where -gen contains `global-isel' . All your potential code generation goes through the DAG.

Reply:
BOLT has no code generation.

Reply:
So we're good to remove the entire block in `bolt/lib/Target/Aarch64/CMakeLists.txt` in the `if(BOLT_BUILD_STANDALONE)`?

Reply:
> So we're good to remove the entire block [...]?

No; what the above comments are saying is that _some_ of these tablegen invocations are unnecessary.

However, bolt does have some target-specific code living in `bolt/lib/Target/*/*MCPlusBuilder.cpp` which requires these headers. So far as I can see this requires re-running tablegen in the standalone build which is not ideal but seems a pragmatic step towards getting a standalone build working.

These are the set of tablegen invocations which I think are necessary. If you delete the wrong ones you'd expect to get a build failure with missing include files.

```
aarch64 needs:
tablegen(LLVM AArch64GenInstrInfo.inc -gen-instr-info)
tablegen(LLVM AArch64GenRegisterInfo.inc -gen-register-info)
tablegen(LLVM AArch64GenSystemOperands.inc -gen-searchable-tables)

x86:
tablegen(LLVM X86GenInstrInfo.inc -gen-instr-info -instr-info-expand-mi-operand-info=0)
tablegen(LLVM X86GenMnemonicTables.inc -gen-x86-mnemonic-tables -asmwriternum=1)
tablegen(LLVM X86GenRegisterInfo.inc -gen-register-info)
tablegen(LLVM X86GenSubtargetInfo.inc -gen-subtarget)

riscv:
tablegen(LLVM RISCVGenInstrInfo.inc -gen-instr-info)
tablegen(LLVM RISCVGenRegisterInfo.inc -gen-register-info)
```

Reply:
Alright, I finally found the time and reduced it down to that list.